### ISSUE TITLE
Converting HF model to GGUF giving error Missing tokenizer.model [Solved]

### ISSUE THREAD

**Issue by overtunned** [ggerganov/llama.cpp#6419]
I saved a fine tuned model using hugging face libraries and was trying to convert it into GGUF when I faced with a error

`Error: Missing tokenizer.model`

But my directory structure had all the required files

```
config.json
generation_config.json
model-00001-of-00003.safetensors
model-00002-of-00003.safetensors
model-00003-of-00003.safetensors
model.safetensors.index.json
special_tokens_map.json
tokenizer_config.json
tokenizer.json
```

The error was cause due to a `sys.exit(1)` command after checking if the tokenizer.model file exits or not. 

Changing it to 
`raise FileNotFoundError(f"File not found: {tokenizer_path}")`

So, the resulting code will look like this

```
        if not tokenizer_path.is_file():
            print(f'Error: Missing {tokenizer_path}', file=sys.stderr)
            raise FileNotFoundError(f"File not found: {tokenizer_path}")
```

Solved the issue

https://github.com/ggerganov/llama.cpp/blob/c50a82ce0f71558cbb8e555146ba124251504b38/convert-hf-to-gguf.py#L327

Edit:

The above issue is a part of this try except block. It should have raised the FileNotFoundError for it to trigger the exception block.

https://github.com/ggerganov/llama.cpp/blob/f87f7b898651339fe173ddf016ca826163e899d8/convert-hf-to-gguf.py#L1201

**Comment by Jeximo** [comment]
> Solved the issue

Based on the **content** of your directory, there's no `tokenizer.model`.  You may have bypassed the error, but I think you excluded `tokenizer.model` from the conversion.

**Comment by overtunned** [comment]
Oh actually this section is part of another try and except where it checks for the tokenizer.model file if not found its proceeds to use the tokenizer.json file. But due to the `sys.exit(1)` the `except FileNotFoundError` was not able to catch the exception since it exited the program instead of return back to the except.

**Comment by Jeximo** [comment]
> if not found its proceeds to use the tokenizer.json file

Thanks for explaining.  I don't know that `tokenizer.json` = `tokenizer.model` in all cases(*it may be, I'm genuinely uncertain*).

I suggest making a pull request, and maintainers may add your contribution after review.

**Comment by overtunned** [comment]
It's not the same I believe. There are certain models that use tokenizer.model but Huggingface transformers I have seen use tokenizer.json

**Comment by cebtenzzre** [comment]
Fixed by #6443
